### PR TITLE
MNT: Python version update, contributors, code of conduct

### PR DESCRIPTION
**Pull request by zbruick**
Addresses 7/8 issues in #270.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # After changing this file, check it on:
 #   http://lint.travis-ci.org/
 language: python
-sudo: false
+sudo: falsei
```
Comment (dopplershift) [inline .travis.yml]: Looks like a typo?

File: .travis.yml [hunk 2]
```diff
@@ -35,14 +37,15 @@ env:
 
 matrix:
   include:
-    - python: 3.5
-      env:
     - python: "3.6-dev"
       env: PRE="--pre"
+    - python: "3.7-dev"
```
Comment (dopplershift) [inline .travis.yml]: Go ahead and replace 3.6-dev with 3.7-dev.

**Comment by zbruick** [comment]
Updated based on the review and rebased on master.

**Comment by zbruick** [comment]
Travis is failing on the same issue as #269. @lesserwhirls mentioned to me that Unidata will be adopting a broader Code of Conduct to be implemented soon(?), so we may not want to add that right now for sake of clarity on this issue. I can take that commit out if desired.

**Comment by lesserwhirls** [comment]
@ddirks is working with UCAR on some of the language used in their Code of Conduct for contributors, but we think that will be resolved in a few days. I have a PR over on Unidata/.github (Unidata/.github#1) that will turn on a default Code of Conduct for the Unidata organization that follows the UCAR Code of Conduct. The UCAR Code of Conduct for contributors is based on the Contributor Covenant version 1.4, so the on in this PR (and on metpy) is very similar, but not identical.

**Comment by dopplershift** [comment]
Let's take out the code of conduct here and just rely on the organization one, since that makes the most sense.

**Comment by zbruick** [comment]
The code of conduct has been dropped from this PR. Travis will probably error out on IGRA, but the rest of the tests should be green.

**Comment by dopplershift** [comment]
Test failures on Travis due to IGRA2 crapping out docs. Merging...

**Pull request merged**